SumoLogic.Logging.NLog/BufferedSumoLogicTarget.cs

```diff
@@ -385,11 +392,12 @@ protected override void FlushAsync(global::NLog.Common.AsyncContinuation asyncCo
                 var task = this.flushBufferTask;
                 if (task != null)
                 {
-                    // i don't love this, but it maintains parity w/ existing sync calls
-                    task.FlushAndSend().GetAwaiter().GetResult();
+                    task.FlushAndSend().ContinueWith(t => asyncContinuation(t.Exception));

```

Comment:
Can this method execute after the `CloseTarget` method? Are we at a risk of losing messages?

Reply:
NLog Flush will wait for the `asyncContinuation`-callback to be called before closing down.

Reply:
Looking at the source code of `NLog` (https://github.com/NLog/NLog/blob/3bced3c2d56bdf5fcc9d530294f83e5981ee47df/src/NLog/Targets/Target.cs), I don't see any relation between `CloseTarget` getting called, and `FlushAsync` getting called. Can you please confirm?

Reply:
When NLog performs shutdown then it resets the active configuration by flushing AND closing. See https://github.com/NLog/NLog/blob/4d207883a91c566a873cbd3c267fa7b7ba815dc0/src/NLog/LogFactory.cs#L816